src/sorted_vector.jl

```diff
@@ -0,0 +1,52 @@
+"""
+A `SortedVector` behaves like a standard Julia `Vector`, except that its elements are stored in sorted order, with an optional function `by` that determines the sorting order in the same way as `Base.searchsorted`.
+"""
+immutable SortedVector{T, F<:Function}
+    data::Vector{T}
+    by::F
+
+    function SortedVector(data::Vector{T}, by::F)
+        new(sort(data), by)
```

Comment:
`rev` isn't redundant with `by`: `sort` takes  both arguments. Would be better to make them keyword arguments, to match `sort`.

It would also be useful to provide a way to sort the data in-place, and to declare that the input is already sorted. Not sure what's the best way to do this; maybe just a `sorted` keyword argument.